src/pyhf/cli/cli.py

```diff
@@ -31,3 +39,18 @@ def pyhf():
 pyhf.add_command(infer.cls)
 
 pyhf.add_command(patchset.cli)
+
+try:
```

Comment:
should we escape it like this? or have `pyhf shellcomplete` always be there but then say sth like `Install pyhf[shellcomplete]` in case the import fails?

also should this be in a separate file?

Reply:
separate file I think. A cleaner variant might be

```python
try:
    import click_completion

    click_completion.init()

    from . import shellcomplete
    pyhf.add_command(shellcomplete.cli)

except ImportError:
    pass
```

or something...

Reply:
done